File: demo-vault/code/hello.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ThinkingKity.Demo;

public enum Priority { Low, Medium, High }

public record Task(
    string Title,
    Priority Priority,
    bool Completed = false
);

public class TaskManager
{
    private readonly List<Task> _tasks = new();

    public void Add(string title, Priority priority) =>
        _tasks.Add(new Task(title, priority));

    public void Complete(string title)
    {
        var index = _tasks.FindIndex(t => t.Title == title);
        if (index >= 0)
            _tasks[index] = _tasks[index] with { Completed = true };
    }

    public IEnumerable<Task> PendingByPriority() =>
        _tasks
            .Where(t => !t.Completed)
            .OrderByDescending(t => t.Priority);

    public void PrintSummary()
    {
        var groups = _tasks
            .Where(t => !t.Completed)
            .GroupBy(t => t.Priority)
            .Select(g => new { Priority = g.Key, Count = g.Count() });

        Console.WriteLine("Pending tasks by priority:");
        foreach (var g in groups)
            Console.WriteLine($"  {g.Priority}: {g.Count}");
    }

    public static void Main()
    {
        var tm = new TaskManager();
        tm.Add("Write unit tests", Priority.High);
        tm.Add("Update README", Priority.Medium);
        tm.Add("Fix typos in docs", Priority.Low);
        tm.Add("Refactor auth module", Priority.High);
        tm.Complete("Update README");

        tm.PrintSummary();

        Console.WriteLine("\nSorted pending:");
        foreach (var t in tm.PendingByPriority())
            Console.WriteLine($"  [{t.Priority}] {t.Title}");
    }
}

File: demo-vault/code/hello.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <map>

enum class Priority { Low, Medium, High };

struct Task {
    std::string title;
    Priority priority;
    bool completed;

    std::string priorityStr() const {
        switch (priority) {
            case Priority::Low:    return "Low";
            case Priority::Medium: return "Medium";
            case Priority::High:   return "High";
        }
        return "Unknown";
    }
};

class TaskManager {
    std::vector<Task> tasks;

public:
    void add(const std::string& title, Priority p) {
        tasks.push_back({title, p, false});
    }

    void complete(size_t index) {
        if (index < tasks.size()) tasks[index].completed = true;
    }

    std::vector<const Task*> pendingByPriority() const {
        std::vector<const Task*> pending;
        for (const auto& t : tasks) {
            if (!t.completed) pending.push_back(&t);
        }
        std::sort(pending.begin(), pending.end(), [](const Task* a, const Task* b) {
            return static_cast<int>(a->priority) > static_cast<int>(b->priority);
        });
        return pending;
    }

    void printSummary() const {
        std::map<Priority, int> counts;
        for (const auto& t : tasks) {
            if (!t.completed) counts[t.priority]++;
        }
        std::cout << "Pending tasks by priority:\n";
        for (auto& [pri, count] : counts) {
            std::cout << "  " << tasks[0].priorityStr() << ": " << count << "\n";
        }
    }
};

int main() {
    TaskManager tm;
    tm.add("Write unit tests", Priority::High);
    tm.add("Update README", Priority::Medium);
    tm.add("Fix typos in docs", Priority::Low);
    tm.add("Refactor auth module", Priority::High);
    tm.complete(1);

    tm.printSummary();

    auto pending = tm.pendingByPriority();
    std::cout << "\nSorted pending:\n";
    for (const auto* t : pending) {
        std::cout << "  [" << t->priorityStr() << "] " << t->title << "\n";
    }

    return 0;
}
